### PR TITLE
add dirty data to job information class

### PR DESCRIPTION
**Pull request by halkeye**
Problem:

Jenkins and Bamboo pass data between handlers, and do some dirty hacks to attempt to determine if data has changed and should issue a save request to the server.

Solution:

Simple solution by adding a hash to track dirty data so we know what to issue an update for.

Reviewers:

@moizjv @mehmetg @mehmetg-sauce 

More changes are coming to the various plugins to use this new updated code.

### CODE REVIEW

File: src/test/java/com/saucelabs/ci/JobInformationTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+public class JobInformationTest {
+
+    @Test
+    public void testPopulateFromJson() throws Exception {
+
+        JSONObject obj = new JSONObject(
+            IOUtils.toString(getClass().getResourceAsStream("/job_info.json"), "UTF-8")
+        );
+        JobInformation job = new JobInformation("1234","hmac");
```
Comment (moizjv) [inline src/test/java/com/saucelabs/ci/JobInformationTest.java]: `JobInformation job = new JobInformation("1234", "hmac");`

Comment (halkeye) [inline src/test/java/com/saucelabs/ci/JobInformationTest.java]: i'm surprised my ide doesn't just fix it

File: src/main/java/com/saucelabs/ci/JobInformation.java [hunk 2]
```diff
@@ -24,87 +26,183 @@
...
     private String version;
 
-
-    private transient boolean hasBuildNumber = true;
-    private transient boolean hasJobName = false;
     private String videoUrl;
     private String logUrl;
+    private String build;
+
+    final private HashMap<String, Object> changes = new HashMap<String, Object>();
```
Comment (moizjv) [inline src/main/java/com/saucelabs/ci/JobInformation.java]: should this go into constructor?

Comment (halkeye) [inline src/main/java/com/saucelabs/ci/JobInformation.java]: why?

Comment (moizjv) [inline src/main/java/com/saucelabs/ci/JobInformation.java]: I think it is just a styling preference. I am fine either ways :)

Comment (halkeye) [inline src/main/java/com/saucelabs/ci/JobInformation.java]: ah, I prefer this way,and most of the other code is done that way.

**Comment by moizjv** [comment]
Cool looks good :+1:

**Comment by mehmetg** [comment]
Looks good to me! :+1: 
Off topic: This is sharing some functionality with sauce_rest doesn't it?

**Comment by halkeye** [comment]
yep, eventually I think most, or all of the non sauceconnect stuff should
move to saucerest

On Wed, Feb 10, 2016 at 5:22 PM, Mehmet <notifications@github.com> wrote:

> Looks good to me! [image: :+1:]
> Off topic: This is sharing some functionality with sauce_rest doesn't it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saucelabs/ci-sauce/pull/15#issuecomment-182666756>.
>



-- 


*Gavin MoganSoftware Developer*

Sauce Labs <http://saucelabs.com/> | Sauce blog <http://sauceio.com/>

Talk to us anytime:  <https://twitter.com/saucelabs>
<https://www.linkedin.com/company/sauce-labs>
<https://plus.google.com/+SauceLabs_Official/posts>
<https://www.facebook.com/saucelabs>

**Pull request merged**